core/src/options/shape.rs

```diff
@@ -12,33 +12,33 @@ use crate::{Error, FromMeta, Result};
 /// variants.
 ///
 /// # Usage
-/// Because `Shape` implements `FromMeta`, the name of the field where it appears is
-/// controlled by the struct that declares `Shape` as a member. That field name is
+/// Because `ShapeOpt` implements `FromMeta`, the name of the field where it appears is
+/// controlled by the struct that declares `ShapeOpt` as a member. That field name is
 /// shown as `ignore` below.
 ///
 /// ```rust,ignore
 /// #[ignore(any, struct_named, enum_newtype)]
 /// ```
 #[derive(Debug, Clone)]
-pub struct Shape {
+pub struct ShapeOpt {
```

Comment:
This is not a good name.

Reply:
I'm not sure, but altering the name might make earlier versions incompatible.?

Reply:
This struct isn't visible outside `darling_core`, so that isn't a risk. The type is a bit weird because it's basically smashing two macro-time `ShapeSet` values together, along with the `any` field. I decided to go with a name based on how it's used, and called it `DeriveInputShapeSet`.